### PR TITLE
aws-lambda | add CodeCommit trigger event

### PR DESCRIPTION
**Pull request by tulsidas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [here](https://docs.aws.amazon.com/lambda/latest/dg/services-codecommit.html)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/aws-lambda/trigger/codecommit.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    codecommit: {
+        references: Array<
+        {
+            commit: string;
+            created?: boolean;
+            deleted?: boolean;
+            ref: string;
+        }>;
+    };
+    customData: string;
```
Comment (jamesorlakin) [inline types/aws-lambda/trigger/codecommit.d.ts:20]: I'm wondering whether this one should be optional given it's not in the example event in the AWS docs, unless it's set to an empty string constantly if a value isn't provided whilst making the event? https://docs.aws.amazon.com/codecommit/latest/userguide/how-to-notify-lambda.html

```suggestion
    customData?: string;
```

(AWS documentation can be quite sparse on specifics so I might be wrong!)

**Review by jamesorlakin — COMMENTED** [review]
Otherwise LGTM, thank you!

File: types/aws-lambda/trigger/codecommit.d.ts (hunk 1 above)
Comment (tulsidas) [reply types/aws-lambda/trigger/codecommit.d.ts:20]: In the tests I made the incoming event always has `customData: ""` even though I configured the trigger with empty custom data field, so I assume it's always present. I couldn't find any specification on that event format, alas....

Comment (tulsidas) [reply types/aws-lambda/trigger/codecommit.d.ts:20]: I can still change it to optional if it suits you better

Comment (jamesorlakin) [reply types/aws-lambda/trigger/codecommit.d.ts:20]: I think optional might be better, cheers! I had a quick play and got customData as a JSON null in my account:
```
{
  "Records": [
    {
      "awsRegion": "eu-west-2",
      "codecommit": {
        "references": [
          {
            "commit": "25f945a6135b70ffb93b3476ed1615f157b7c011",
            "ref": "refs/heads/main"
          }
        ]
      },
      "customData": null,
      "eventId": "72f89f41-d36c-49e0-9719-e068b2284556",
      "eventName": "ReferenceChanges",
      "eventPartNumber": 1,
      "eventSource": "aws:codecommit",
      "eventSourceARN": "arn:aws:codecommit:eu-west-2:869036666965:test-repo",
      "eventTime": "2023-03-13T18:15:49.572+0000",
      "eventTotalParts": 1,
      "eventTriggerConfigId": "4037999f-048a-4ff3-a479-d7f0fd8b6671",
      "eventTriggerName": "TestTrigger",
      "eventVersion": "1.0",
      "userIdentityARN": "arn:aws:sts::869036666965:assumed-role/AWSReservedSSO_AWSAdministratorAccess_blah"
    }
  ]
}
```

**Review by jamesorlakin — APPROVED** [review]
Thanks @tulsidas

**Review by jakebailey — APPROVED** [review]

**Comment by tulsidas** [comment]
Ready to merge

**Pull request merged**